src/PureScript/Backend/Chez/Syntax.purs

```diff
@@ -103,10 +103,14 @@ definitionIdentifiers :: ChezDefinition -> Prim.Array Prim.String
 definitionIdentifiers (DefineValue i _) = [ i ]
 definitionIdentifiers (DefineCurriedFunction i _ _) = [ i ]
 definitionIdentifiers (DefineUncurriedFunction i _ _) = [ i ]
-definitionIdentifiers (DefineRecordType i _) =
+definitionIdentifiers (DefineRecordType i []) =
   [ recordTypeUncurriedConstructor i
   , recordTypePredicate i
   ]
+definitionIdentifiers (DefineRecordType i fields) =
+  [ recordTypeUncurriedConstructor i
+  , recordTypePredicate i
+  ] <> map (recordTypeAccessor i) fields
```

Comment:
I think matching against an empty array is redundant here? Unless there's some extra code you want to add when it's empty.
```suggestion
definitionIdentifiers (DefineRecordType i fields) =
  [ recordTypeUncurriedConstructor i
  , recordTypePredicate i
  ] <> map (recordTypeAccessor i) fields
```